server/internal/adapter/internalapi/server.go

```diff
@@ -0,0 +1,143 @@
+package internalapi
+
+import (
+	"context"
+
+	"github.com/reearth/reearth/server/internal/adapter"
+	"github.com/reearth/reearth/server/internal/adapter/internalapi/internalapimodel"
+	pb "github.com/reearth/reearth/server/internal/adapter/internalapi/schemas/internalapi/v1"
+	"github.com/reearth/reearth/server/internal/usecase/interfaces"
+	"github.com/reearth/reearth/server/pkg/id"
+	"github.com/reearth/reearth/server/pkg/visualizer"
+	"github.com/reearth/reearthx/account/accountdomain"
+)
+
+type server struct {
+	pb.UnimplementedReEarthVisualizerServer
+}
+
+func NewServer() pb.ReEarthVisualizerServer {
+	return &server{}
+}
+
+func (s server) GetProjectList(ctx context.Context, req *pb.GetProjectListRequest) (*pb.GetProjectListResponse, error) {
```

Comment:
```suggestion
func (s server) GetProjectList(ctx context.Context, req *pb.GetProjectListRequest) (*pb.GetProjectListResponse, error) {
```

We'd like to add the provision to get list project when the user authenticated or not. i.e
authenticated -> All projects
not-authenticated -> Public Projects

Reply:
Please address the above on the schema as well.

Reply:
Owner(workspace) can only change the visibility of the project.

GetProjectList (not authenticated) -> Return all the public projects that the user is part of.

Reply:
(Authenticated) User Alias -> User -> (Workspace)*n -> All the Projects

(Not Authenticated) User Alias -> User -> (Workspace)*n -> All Public Projects

Reply:
yes fixed